File: Medium/Array/3689. Maximum Total Subarray Value I.cpp
// 3689. Maximum Total Subarray Value I
class Solution {
public:
    long long maxTotalValue(vector<int>& nums, int k) {

        int max_val = nums[0];
        int min_val = nums[0];

        for(auto i : nums){
            max_val = max(max_val,i);
            min_val = min(min_val,i);
        }
        long long ans = max_val - min_val;

        return (ans*k);
        
    }
};

File: Medium/Array/3689. Maximum Total Subarray Value I.py
# 3689. Maximum Total Subarray Value I
class Solution:
    def maxTotalValue(self, nums: List[int], k: int) -> int:

        mx = nums[0]
        mn = nums[0]

        for i in nums:
            mx = max(mx,i)
            mn = min(mn,i)

        return (mx - mn)*k
        
